pkg/controllers/sgx/controller.go

```diff
@@ -117,20 +114,7 @@ func setInitContainer(spec *v1.PodSpec, imageName string) {
 func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	devicePlugin := rawObj.(*devicepluginv1.SgxDevicePlugin)
 
-	var nodeSelector map[string]string
-	dpNodeSelectorSize := len(devicePlugin.Spec.NodeSelector)
-	if dpNodeSelectorSize > 0 {
-		nodeSelector = make(map[string]string, dpNodeSelectorSize+1)
-		for k, v := range devicePlugin.Spec.NodeSelector {
-			nodeSelector[k] = v
-		}
-		nodeSelector["kubernetes.io/arch"] = amd64
-	} else {
-		nodeSelector = map[string]string{"kubernetes.io/arch": amd64}
-	}
-
 	daemonSet := deployments.SGXPluginDaemonSet()
-	daemonSet.Spec.Template.Spec.NodeSelector = nodeSelector
```

Comment:
I don't think you can drop these lines. If the user has specified `nodeSelector` in `SgxDevicePlugin`, the daemonset will not have `nodeSelector`added.

Reply:
Ah, I see. But, some other plugins (qat, fpga) didn't have it in the first place though there has been a nodeselector field in the sample yaml file.! Is it just a mistake or is there another reason?

Reply:
As we include node selector into all daemonset yamls we may want to go back to having an API and calling it from  NewDaemonSet and UpdateDaemonSet for all plugins.

Reply:
@hj-johannes-lee it's a bug

> may want to go back to having an API and calling it from NewDaemonSet and UpdateDaemonSet for all plugins

+1

Reply:
We can also replace node selector if plugin has it. This would be simply something like this:
```
if len(devicePlugin.Spec.NodeSelector) > 0 {
        daemonSet.Spec.Template.Spec.NodeSelector = devicePlugin.Spec.NodeSelector
}
```
@mythi would this be enough?

Reply:
It's what I am doing now, but can I ask why do we need to count the length of devicePlugin.Spec.NodeSelector? 
Can it not just be 
>        daemonSet.Spec.Template.Spec.NodeSelector = devicePlugin.Spec.NodeSelector

?

Reply:
@bart0sh  I was doing the same, but without the if clause. I wanted to ask why we need, but it would be faster to just do as you said and listen later.

Reply:
without it default node selector value taken from daemonset yaml would be unconditionally replaced even by empty node selector from the plugin. This would make your changes in daemonset yamls useless and they'll be always replaced.

Reply:
@hj-johannes-lee the same is true for the `UpdateDaemonSet()` path, i.e., ```if len(devicePlugin.Spec.NodeSelector) == 0```  -> restore the default

Reply:
Testing this either in the controller_test or in the envtest would not harm either.

Reply:
yes, I created #772 for that